snowflake/src/integration/java/org/apache/iceberg/snowflake/NamespaceTests.java

```diff
@@ -0,0 +1,258 @@
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"Schema", "schema", "_schema", "Schema123", "schema$", "_Schema$_123"})
+  public void testLoadNamespaceWithValidUnquotedIdentifier(String schema)
+      throws SQLException, InterruptedException {
+    String dbName = getDatabase();
+    try {
+      clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+      createOrReplaceSchema(schema);
+      Assertions.assertThat(
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, schema)).isEmpty())
+          .isTrue();
+    } finally {
+      dropSchemaIfExists(schema);
+    }
+  }
+
+  @Test
+  public void testLoadNamespaceWithUnquotedIdentifierIsCaseInsensitive()
+      throws SQLException, InterruptedException {
+    String schema = "Schema123";
+    String dbName = getDatabase();
+    try {
+      clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+      createOrReplaceSchema(schema);
+      Assertions.assertThat(
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, schema)).isEmpty())
+          .isTrue();
+      Assertions.assertThat(
+              snowflakeCatalog
+                  .loadNamespaceMetadata(Namespace.of(dbName, schema.toUpperCase()))
+                  .isEmpty())
+          .isTrue();
+      Assertions.assertThat(
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, "schEmA123")).isEmpty())
+          .isTrue();
+    } finally {
+      dropSchemaIfExists(schema);
+    }
+  }
+
+  @Test
+  public void testLoadNamespaceWithQuotedIdentifierIsCaseSensitive()
+      throws SQLException, InterruptedException {
+    String schema = "\"Schema123\"";
+    String dbName = getDatabase();
+    try {
+      clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+      createOrReplaceSchema(schema);
+      Assertions.assertThatThrownBy(
+              () ->
+                  snowflakeCatalog.loadNamespaceMetadata(
+                      Namespace.of(dbName, schema.toUpperCase())))
+          .isInstanceOf(NoSuchNamespaceException.class)
+          .hasMessageContainingAll("snowflake identifier", "doesn't exist");
+    } finally {
+      dropSchemaIfExists(schema);
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(
+      strings = {
+        "\"Schema123\"",
+        "\"MySchema\"",
+        "\"my.schema\"",
+        "\"my schema\"",
+        "\"My 'Schema'\"",
+        "\"3rd_schema\"",
+        "\"$Ischema\"",
+        "\"H@!!.0-w*r()^'\"",
+        "\"идентификатор\""
+      })
+  public void testLoadNamespaceWithQuotedIdentifier(String schema)
+      throws SQLException, InterruptedException {
+    String dbName = getDatabase();
+    try {
+      clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+      createOrReplaceSchema(schema);
+      Assertions.assertThat(
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, schema)).isEmpty())
+          .isTrue();
```

Comment:
it should be `Assertions.assertThat(snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, schema))).isEmpty()` where the `isEmpty()` call is on the outer `assertThat(..)` ->  `assertThat(..).isEmpty()`

Reply:
Thanks @nastra! Switched to the recommendation.